src/parquet_writer.h

```diff
@@ -0,0 +1,190 @@
+#ifndef PARQUET_WRITER_H
+#define PARQUET_WRITER_H
+
+#include <napi.h>
+
+#include <parquet/arrow/writer.h>
+#include <parquet/stream_writer.h>
+
+#include <vector>
+
+struct Column {
+  std::string key;
+  parquet::Type::type type;
+};
+
+class ParquetWriter : public Napi::ObjectWrap<ParquetWriter> {
+public:
+  static Napi::Object Init(Napi::Env env, Napi::Object exports) {
+    Napi::Function func =
+      DefineClass(env,
+        "ParquetWriter", {
+          InstanceMethod("appendRow",    &ParquetWriter::appendRow),
+          InstanceMethod("open", &ParquetWriter::open),
+          InstanceMethod("setRowGroupSize", &ParquetWriter::setRowGroupSize)
+        });
+
+    auto constructor = new Napi::FunctionReference();
+    *constructor = Napi::Persistent(func);
+    env.SetInstanceData(constructor);
+
+    exports.Set("ParquetWriter", func);
+    return exports;
+  }
+
+public:
+  ParquetWriter(const Napi::CallbackInfo& info)
+    : Napi::ObjectWrap<ParquetWriter>(info)
+    , builder()
+  {
+    auto env = info.Env();
+    if (info.Length() < 2 || !info[0].IsObject() || !info[1].IsString()) {
+      Napi::TypeError::New(env, "schema::Object, path::string expected").ThrowAsJavaScriptException();
+      return;
+    }
+
+    filepath = info[1].ToString().Utf8Value();
+
+    // Build schema
+    auto schema_obj = info[0].As<Napi::Object>();
+    auto keys = schema_obj.GetPropertyNames();
+    parquet::schema::NodeVector fields;
+    for (uint32_t i = 0; i < keys.Length(); i++) {
+      auto name = keys.Get(i).ToString().Utf8Value();
+      auto type_name = schema_obj.Get(name).ToObject().Get("type").ToString().Utf8Value();
+      auto converted_type = convertedTypeFromString(type_name);
+      auto logical_type = logicalTypeFromString(type_name);
+      columns.push_back(Column{name, logical_type});
+      fields.push_back(parquet::schema::PrimitiveNode::Make(
+        name,
+        parquet::Repetition::OPTIONAL,
+        logical_type,
+        converted_type));
+    }
+    schema = std::static_pointer_cast<parquet::schema::GroupNode>(
+      parquet::schema::GroupNode::Make("schema", parquet::Repetition::OPTIONAL, fields)
+    );
+  }
+
+  Napi::Value appendRow(const Napi::CallbackInfo& info) {
+    auto env = info.Env();
+    if (info.Length() < 1 || !info[0].IsObject()) {
+      Napi::TypeError::New(env, "row::Object expected").ThrowAsJavaScriptException();
+      return Napi::Boolean::New(env, false);
```

Comment:
`env.Null()`, don't return false. The thrown error already encodes the failure, no need for a boolean here (that will never be read anyway, due to the thrown exception).